src/graph.cpp

```diff
@@ -348,10 +370,10 @@ void Graph::write_binary(const std::string& filename) {
   // after this point all updates have been processed from the buffering system
 
   auto binary_out = std::fstream(filename, std::ios::out | std::ios::binary);
-  int fail_factor = Sketch::get_failure_factor();
-  binary_out.write((char*)&seed, sizeof(long));
-  binary_out.write((char*)&num_nodes, sizeof(uint64_t));
-  binary_out.write((char*)&fail_factor, sizeof(int));
+  auto fail_factor = Sketch::get_failure_factor();
```

Comment:
This makes me nervous. I suppose if it works than it works though.

Reply:
what part of it?

Reply:
Using auto for a variable we're going about to write to disk. I guess I'm nervous the compiler might chose some weird larger or smaller type to put fail_factor in. Though I don't know if that's actually possible.

Reply:
auto takes the type of the RHS at compile time. it’s like if we looked up the type of get_failure_factor (uint32) and spliced it in

Reply:
Gotcha, sounds good